Detector.cc

```
@@ -4085,7 +4108,204 @@ This function has two main parts: (1) loading of gain/phase values from gainFile
 }
 
 
+inline void Detector::ReadTrig_Delays_Masking(string filename, Settings *settings1) { //will return trigger delays and channels masked from trigger logic
+
+	
+	//First, let's check that the file exists and the format is correct
+	
+	// check if the file with delays + masking exists
+	char errorMessage[400];
+        struct stat buffer;
+
+        bool trigFileExists = (stat(filename.c_str(), &buffer)==0);
+        if (!trigFileExists){
+		sprintf(errorMessage, "Trigger formation file is not found (trigger file exists %d) ", trigFileExists);
+        	throw std::runtime_error(errorMessage);
+        }
+
+	ifstream trigFile(filename.c_str()); // open the file
+	string line; // a dummy variable we can stream over
+
+	// if the trigger formation file exists, then make sure our user has formatted the file correctly
+	// in particular, it means we really need to see the word "Channel No." as the first word in the header file 
+
+	string expected_first_column_header = "Channel";
+        if(trigFile.is_open()){
+                while(trigFile.good()){

```

is this while loop necessary? if im reading it right it seems like it should either throw an error or break the loop on the first iteration? maybe it would be more clear what the code is doing if we remove the loop if it is superfluous